### PR TITLE
Idp select UI

### PR DESCRIPTION
**Pull request by dan-f**
This PR introduces a popup-based workflow for logging in to solid IDPs, shielding client applications from redirects.

- [x] demo working
- [x] tests and refactor
- [x] create build scripts for idp select / redirect app
- [x] optimize the builds
- [x] style the popup app
- [x] simplify .env files
- [x] remove *.bundle.js from the inlined html bundle directories
- [x] add text box for logging in to a "custom" IDP
- [x] report errors to the user in the popup

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,17 +3,30 @@
...
 [WebID-TLS](https://www.w3.org/2005/Incubator/webid/spec/tls/) and
 [WebID-OIDC](https://github.com/solid/webid-oidc-spec).
 
 This library abstracts away the implementation details of these specs so that
 clients don't have to handle different authentication protocols.
 
+### Why might I need this?
+
+If you're building a web app and want to identify users with Solid, or store
+personal information on your user's solid account, you'll have to authenticate
```
Comment (RubenVerborgh) [inline README.md]: *Solid

File: README.md [hunk 2]
```diff
@@ -3,17 +3,30 @@
...
+them.  This library provides a simple API for logging in, logging out, and
+fetching resources with authenticated credentials.
+
+### How can I use this?
+
+The simplest way to use this library is to install it via `npm` or `yarn`.  You can then use the ES6 module (`import { login, currentUser, logout } from 'solid-auth-client'`), or you can grab the transpiled UMD bundle from `node_modules/solid-auth-client/dist-lib/solid-auth-client.bundle.js`.
+
 ## API
```
Comment (RubenVerborgh) [inline README.md]: Takes a bit of time here until we get to the meat. Could it make sense to put the types at the end?

File: README.md [hunk 3]
```diff
@@ -119,6 +132,57 @@ authenticate scheme.
...
+#### Configuration
+
+The demo app is configurable via the `.env.demo` file.  The important fields are:
+
+- IDP_SELECT_URI: URI for the popup-based IDP select app.  When testing
+  locally, this will be something like 'http://localhost:XXXX/idp-select.html'
+- CALLBACK_URI: URI for the popup-based callback app.  When testing locally,
+  this will be something like 'http://localhost:XXXX/idp-callback.html'
+
+### Building the popup app
```
Comment (RubenVerborgh) [inline README.md]: Perhaps explain why I would want to do this?

File: README.md [hunk 4]
```diff
@@ -119,6 +132,57 @@ authenticate scheme.
...
+```sh
+$ git clone https://github.com/solid/solid-auth-client.git
+$ cd solid-auth-client
+$ yarn
+$ yarn build # build the library and UMD bundle
+$ yarn test # run the code formatter, linter, and test suite
+$ yarn test:dev # just run the tests in watch mode
+```
+
+### Building the demo app
```
Comment (RubenVerborgh) [inline README.md]: Shouldn't the popup come first?

File: README.md (hunk 3 above)
Comment (RubenVerborgh) [reply README.md]: Maybe "app" is a bit confusing, in the sense that "demo app" and "popup app" mean different things. The popup is a mandatory component, the demo is just an external app.

File: README.md [hunk 5]
```diff
@@ -3,17 +3,30 @@
 [![Build Status](https://travis-ci.org/solid/solid-auth-client.svg?branch=master)](https://travis-ci.org/solid/solid-auth-client)
 [![Coverage Status](https://coveralls.io/repos/github/solid/solid-auth-client/badge.svg?branch=master)](https://coveralls.io/github/solid/solid-auth-client?branch=master)
 
-Opaquely authenticates solid clients
+Opaquely authenticates [Solid](https://github.com/solid/) clients
 
 ## About
```
Comment (RubenVerborgh) [inline README.md]: Very clear!

File: .env.demo.example [hunk 6]
```diff
@@ -0,0 +1,2 @@
+IDP_SELECT_URI=http://localhost:8081/idp-select.html
+CALLBACK_URI=http://localhost:8081/idp-callback.html
```
Comment (RubenVerborgh) [inline .env.demo.example]: How about 8443 as default port (to align with node-solid-server)?

File: README.md [hunk 7]
```diff
@@ -84,17 +117,94 @@ logout (storage?: Storage): Promise<void>
...
+- IDP_SELECT_URI: URI for the popup-based IDP select app.  When testing
+  locally, this will be something like 'http://localhost:XXXX/idp-select.html'
+- CALLBACK_URI: URI for the popup-based callback app.  When testing locally,
+  this will be something like 'http://localhost:XXXX/idp-callback.html'
+
+### Building the popup app
+
+```sh
+$ cp .env.popup.example .env.popup && $EDITOR .env.popup # configure the popup app
+$ yarn start:popup
```
Comment (RubenVerborgh) [inline README.md]: I was confused here that this didn't just give me the HTML files I need, but instead starts an app. (I mean, the "start" part is named well, but it's just not what I expect.)

How can I generate the files?

Comment (RubenVerborgh) [reply README.md]: It's also weird that both the popup and demo are started on 8080.

File: README.md (hunk 3 above)
Comment (RubenVerborgh) [reply README.md]: And the title is about building, but we are actually running it.

File: .env.demo.example (hunk 6 above)
Comment (RubenVerborgh) [reply .env.demo.example]: …and https:// too.

File: .env.popup.example [hunk 8]
```diff
@@ -0,0 +1,3 @@
+TRUSTED_APP_NAME="Your App Here"
```
[the remark this answers is not in the document]
Comment (dan-f) [reply .env.popup.example]: Addressed in the README

File: .env.popup.example [hunk 9]
```diff
@@ -0,0 +1,3 @@
...
+TRUSTED_APP_ORIGIN="http://localhost:8080"
```
[the remark this answers is not in the document]
Comment (dan-f) [reply .env.popup.example]: Actually, this is intended to be the origin of the trusted application.

File: README.md (hunk 5 above)
Comment (dan-f) [reply README.md]: ty :)

File: README.md (hunk 1 above)
Comment (dan-f) [reply README.md]: Coming soon...

File: README.md (hunk 4 above)
Comment (dan-f) [reply README.md]: Yeah, this is addressed now, I hope.

File: demo/components/PersonalInfo.js [hunk 10]
```diff
@@ -2,51 +2,57 @@
...
 
   fetchProfile = (webId: string): Promise<profile> => {
-    const { fetch } = this.props
     const query = `
       @prefix foaf http://xmlns.com/foaf/0.1/
       ${webId} { foaf:name }
     `
-    return fetch('https://databox.me/,query', { method: 'POST', body: query })
-      .then(resp => resp.json())
+    return fetch('https://databox.me/,query', {
```
[the remark this answers is not in the document]
Comment (dan-f) [reply demo/components/PersonalInfo.js]: This is using my live twinql query endpoint :)

**Comment by dan-f** [comment]
@RubenVerborgh - here's the final draft. If there's anything else you'd like to see changed, let me know!

**Pull request merged**